### PR TITLE
Camel Datasonnet Extension

### PR DESCRIPTION
**Pull request by javaduke**

### CODE REVIEW

**Comment by javaduke** [comment]
Hopefully third time is a charm :)

**Review by oscerd — APPROVED** [review]

**Comment by ppalaga** [comment]
```
2022-05-05T04:42:00.9306924Z [INFO] --- license-maven-plugin:3.0:check (license-format) @ camel-quarkus-integration-test-datasonnet ---
2022-05-05T04:42:00.9342263Z [INFO] Checking licenses...
2022-05-05T04:42:00.9446423Z [WARNING] Unknown file extension: /home/runner/work/camel-quarkus/camel-quarkus/integration-tests/datasonnet/src/main/resources/readXMLExtTest.ds
2022-05-05T04:42:00.9447570Z [WARNING] Unknown file extension: /home/runner/work/camel-quarkus/camel-quarkus/integration-tests/datasonnet/src/main/resources/simpleMapping.ds
2022-05-05T04:42:00.9448533Z [WARNING] Unknown file extension: /home/runner/work/camel-quarkus/camel-quarkus/integration-tests/datasonnet/src/main/resources/writeJavaTest.ds
2022-05-05T04:42:00.9450050Z [WARNING] Unknown file extension: /home/runner/work/camel-quarkus/camel-quarkus/integration-tests/datasonnet/src/main/resources/readJavaTest.ds
2022-05-05T04:42:00.9450899Z [WARNING] Unknown file extension: /home/runner/work/camel-quarkus/camel-quarkus/integration-tests/datasonnet/src/main/resources/readCSVTest.ds
```

**Comment by javaduke** [comment]
Something is not right, I did add the mapping for `ds`: 
https://github.com/javaduke/camel-quarkus/blob/datasonnet/pom.xml#L529

**Comment by javaduke** [comment]
Do I need to add anything else besides that `<ds>` tag in mappings section in the main pom so that the license check is successful? It is happening locally too and I don't know how to fix it. Any suggestions?

File: poms/bom/pom.xml [hunk 1]
```diff
@@ -1717,6 +1717,29 @@
                     </exclusion>
                 </exclusions>
             </dependency>
+            <dependency>
+                <groupId>org.apache.camel</groupId>
+                <artifactId>camel-datasonnet</artifactId>
+            <version>${camel.version}</version>
```
Comment (jamesnetherton) [inline poms/bom/pom.xml:1723]: Nit - please fix the indentation.

**Comment by jamesnetherton** [comment]
> Do I need to add anything else besides that `<ds>` tag in mappings section in the main pom so that the license check is successful? It is happening locally too and I don't know how to fix it. Any suggestions?

It's probably because you have a typo in the comment style `SHASHSTAR_STYLE`. It should hopefully work once corrected.

The project version has moved on to `2.10.0-SNAPSHOT`. Please can you update the `pom.xml` versions and also the properties `<camel.quarkus.jvmSince>` & `<camel.quarkus.nativeSince>`. Then rerun the build to rebuild the generated files.

**Comment by javaduke** [comment]
All done

**Comment by jamesnetherton** [comment]
> All done

Please run the build in the `poms` module and commit the regenerated BOM files.

**Comment by javaduke** [comment]
I did run `mvn clean install` in the `poms`, nothing has changed.

**Comment by jamesnetherton** [comment]
I attached the diff that I get. I'm not entirely clear why that happens TBH...

[datasonnet.txt](https://github.com/apache/camel-quarkus/files/8651202/datasonnet.txt)

**Comment by javaduke** [comment]
I see the diff but I'm not sure why affinity is removed, it has nothing to do with datasonnet...

**Comment by jamesnetherton** [comment]
Are you building on a Mac? Could be because of this:

https://github.com/apache/camel-quarkus/blob/69302f8acfcce514f66747cfdde7824f33d827c5/integration-tests/hazelcast/pom.xml#L98-L113

**Comment by javaduke** [comment]
Yeah, I am on Mac. Should I disable this before building poms?

**Comment by javaduke** [comment]
I tried to deactivate profile, `mvn clean install -P \!hazelcast-mac`, didn't help

**Comment by javaduke** [comment]
Done, let's see how it works. 
I'm traveling for the next two days, but will be checking my email periodically, please let me know if there are still any issues.

**Review by jamesnetherton — APPROVED** [review]

**Comment by jamesnetherton** [comment]
Thanks for persevering with this @javaduke!

**Comment by aldettinger** [comment]
yes, nice job @javaduke and many thanks to every helpers. It was not an easy one :)

**Pull request merged**